src/parallel/job_token.rs

```diff
@@ -183,10 +183,30 @@ mod inherited_jobserver {
                     }
                     Ok(None) => YieldOnce::default().await,
                     Err(err) if err.kind() == io::ErrorKind::Unsupported => {
+                        let mut err = None;
                         // Fallback to creating a help thread with blocking acquire
-                        let helper_thread = self
-                            .helper_thread
-                            .get_or_try_init(|| HelperThread::new(self.jobserver))?;
+                        let helper_thread = self.helper_thread.get_or_init(|| {
+                            match HelperThread::new(self.jobserver) {
+                                Ok(thread) => Some(thread),
+                                Err(error) => {
+                                    err = Some(error);
+                                    None
+                                }
+                            }
+                        });
+
+                        if let Some(err) = err {
+                            return Err(err.into());
+                        }
+
+                        let helper_thread = if let Some(thread) = helper_thread {
+                            thread
+                        } else {
+                            return Err(Error::new(
+                                ErrorKind::JobserverHelpThreadError,
+                                "Creation failed",
+                            ));
+                        };
```

Comment:
Why are we changing all this?

Reply:
Because the helper thread creation might fail, and since I can't implement `get_or_try_init` in a few lines, I put an `Option` there instead for the error.

Reply:
if "I can't implement `get_or_try_init` in a few lines" is a concern, we shouldn't have an abstraction at all: we should just put everything right here, no `OnceCell` or `OnceLock` generic type.

adding the type to avoid a cyclic compilation error but not being willing to maintain an equivalent API means that we really should just be using a dependency, whether it's the crate or the stdlib. having the type for a single usage invites adding more usages of it. but if we have more usages added to this crate, but aren't willing to either use a dependency, or maintain an equivalent API, then we are just courting incorrectness creeping in.

Reply:
The problem is we can't call `Once::pison` here.

The stdlib version can access that private method but we can't do it here, so the `OnceLock` impl here will have some paper cuts.

Alternatively, I can implement a `OnceLock::get_or_try_init` that panics, if you try to call it again after it fails.

Reply:
Right, I wouldn't expect us to actively poison it ourselves.

Sure, that sounds fine.